src/System.Text.Encodings.Web/tests/JavaScriptStringEncoderTests.cs

```diff
@@ -232,6 +251,22 @@ public static IEnumerable<object[]> EscapingTestData_NonAscii
                     new object[] { '\'', JavaScriptEncoder.UnsafeRelaxedJsonEscaping, false },
                     new object[] { '+', JavaScriptEncoder.UnsafeRelaxedJsonEscaping, false },
                     new object[] { '\uFFFD', JavaScriptEncoder.UnsafeRelaxedJsonEscaping, false },
+
+                    new object[] { 'a', new MyCustomEncoder(UnicodeRanges.All), false },
+                    new object[] { '\u001F', new MyCustomEncoder(UnicodeRanges.All), true },
+                    new object[] { '\u2000', new MyCustomEncoder(UnicodeRanges.All), true },
+                    new object[] { '\u00A2', new MyCustomEncoder(UnicodeRanges.All), false },
+                    new object[] { '\uA686', new MyCustomEncoder(UnicodeRanges.All), false },
+                    new object[] { '\u6C49', new MyCustomEncoder(UnicodeRanges.All), false },
+                    new object[] { '"', new MyCustomEncoder(UnicodeRanges.All), true },
+                    new object[] { '\\', new MyCustomEncoder(UnicodeRanges.All), true },
+                    new object[] { '<', new MyCustomEncoder(UnicodeRanges.All), true },
+                    new object[] { '>', new MyCustomEncoder(UnicodeRanges.All), true },
+                    new object[] { '&', new MyCustomEncoder(UnicodeRanges.All), true },
+                    new object[] { '`', new MyCustomEncoder(UnicodeRanges.All), true },
+                    new object[] { '\'', new MyCustomEncoder(UnicodeRanges.All), true },
+                    new object[] { '+', new MyCustomEncoder(UnicodeRanges.All), true },
+                    new object[] { '\uFFFD', new MyCustomEncoder(UnicodeRanges.All), false },
```

Comment:
What is the difference between this block of tests and the block added earlier?  I'm having trouble seeing the difference. Seems like there are a bunch of others in the existing cases that are also the same.  Why the duplication?

Reply:
Good point.

`EscapingTestData_NonAscii` is purely a subset of `EscapingTestData`. The only difference is `EscapingTestData` has more cases. I I can use `EscapingTestData_NonAscii` within `EscapingTestData` and get rid of the duplicates.

`EscapingTestData` contains `JavaScriptEncoder.Default` and `JavaScriptEncoder.Create(UnicodeRanges.BasicLatin)` which `EscapingTestData_NonAscii` does not.